codex/codex.nim

```diff
@@ -141,14 +141,18 @@ proc stop*(s: CodexServer) {.async.} =
 
   s.runHandle.complete()
 
-proc new*(T: type CodexServer, config: CodexConf, privateKey: CodexPrivateKey): T =
+
+proc newCodexServer*(
```

Comment:
We use `new` (for ref objects) and `init` (for objects) in many places in the codebase besides just this one. As an alternative to decorating our codebase with these changes everywhere, is this an issue we could potentially contribute to nimsuggest upstream to prevent the crashes?

Reply:
Ah, actually this one isn't directly related to the nimsuggest fix. 

I used it for tracking down the issue as the `new` proc's are a generic overload and nimsuggest won't fully type check them until they're used somewhere. 

For that reason with large complicated it's helpful to split into a concrete proc with a `new` wrapper. 

I could revert it back to `new`, but I think making the `newCodexServer` private would make `new` the external api but provide the compile checking.